articles/150_ros_command_line.md

```diff
@@ -0,0 +1,121 @@
+In ROS 2, this interface had to become more complex to cope with a larger set of configuration options, an ambiguity in remapping rules and parameter assignment syntax (as a result of the leading underscore name convention for hidden resources), a one-to-many relationship between executables and nodes, to name a few.
+
+Because of this, increasingly precise addressing mechanisms as well as leading double underscores (`__`) in some positional arguments, both natural extensions of existing ROS 1 command line features, are combined with ROS 2 specific command line flags.
+Flags, in contrast with other custom syntax alternatives, are:
+
+- Widely known and used.
+- Explicit and less error prone.
+- Suitable for tab completion.
+
+Unfortunately, since these flags coexist with user-defined ones, additional guarding and extraction devices must be put in place -- one of the reasons why these were avoided entirely in ROS 1 command lines.
+
+## Namespacing
+
+To prevent ROS specific command line flags from colliding with user-defined ones, the former are scoped using the `--ros-args` flag and a trailing double dash token (`--`):
+
+```sh
+ros2 run some_package some_node [<user-defined-arg0>...] \
+  --ros-args <ros-specific-arg-0> [<ros-specific-arg-1>...] -- <user-defined-arg-1> [<user-defined-arg-2>...]
+```
+
+If no user defined arguments are provided after ROS specific arguments are, the double dash token (`--`) may be elided:
+
+```sh
+ros2 run some_package some_node [<user-defined-arg-0>...] --ros-args <ros-specific-arg-0> [<ros-specific-arg-1>...]
+```
+
+More than one set of ROS specific flags may appear in the same command line:
+
+```sh
+ros2 run some_package some_node --ros-args <ros-specific-arg-0> -- [<user-defined-arg-0>...] --ros-args <ros-specific-arg-1>
+```
+
+This way, multiple sources, potentially unaware of each other, can append flags to the command line with no regard for previous sets.
+
+## Capabilities
+
+### Summary
+
+As a quick summary of ROS command line capabilities:
+
+- For name remapping, use either `--remap from:=to` or `-r from:=to`.
+- For parameter assignment, use either `--param name:=value` or `-p name:=value`.
+
+### Name remapping rules
+
+Remapping rules may be introduced using the `--remap`/`-r` option.
+This option takes a single `from:=to` remapping rule.
+
+As an example, to remap from `foo` to `bar` upon running `some_node`, one may execute:
+
+```sh
+ros2 run some_package some_node --ros-args --remap foo:=bar
+```
+
+or its shorter equivalent:
+
+```sh
+ros2 run some_package some_node --ros-args -r foo:=bar
+```
+
+### Parameter assignment statements
+
+Parameter assignment may be achieved using the `--param`/`-p` option.
+This option takes a single `name:=value` assignment statement, where `value` is in YAML format.
+
+As an example, to assign an integer value of `2` to an `int_param` upon running `some_node`, one may execute:
```

Comment:
I am not sure if `integer` is a good example here. That also opens the question what makes this a parameter of type since there needs to be some magic to convert the type solely based on its value which has problems on its own.

Therefore I would suggest to simply use a string value here.

Reply:
Hmm, I agree with using a string value to simplify the example but I also think that parameter value syntax cannot be left out of a command line specification. Here, just like when [parameter files are parsed in `rcl`](https://github.com/ros2/rclcpp/blob/8c48dbede7de2bc7e5199f340b6bfd33ffd6a784/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp#L137) and when [`launch` substitutions occur in `Node` parameters](https://github.com/ros2/launch_ros/blob/master/launch_ros/launch_ros/utilities/evaluate_parameters.py#L70), we delegate on YAML format rules for type inference based on values. Whether that's a good call or not is likely worth a separate discussion.